pages/[category]/index.tsx

```diff
@@ -1,51 +1,236 @@
-import { useEffect } from 'react'
+import { useEffect, useState } from 'react'
```

Comment:
this file is very large and I don't think it needs to be. I think that a lot of this can be broken down into components, and that will make it much easier to manage